src/variety/simplegako.js

```diff
@@ -0,0 +1,171 @@
+//
+// パズル固有スクリプト部 シンプルガコ版 simplegako.js
+//
+
+(function(pidlist, classbase) {
+	if (typeof module === "object" && module.exports) {
+		module.exports = [pidlist, classbase];
+	} else {
+		pzpr.classmgr.makeCustom(pidlist, classbase);
+	}
+})(["simplegako"], {
+	//---------------------------------------------------------
+	// マウス入力系
+	MouseEvent: {
+		inputModes: {
+			edit: ["number", "clear"],
+			play: ["number", "clear", "copynum"]
+		},
+
+		mouseinput_other: function() {
+			if (this.inputMode === "copynum") {
+				this.dragnumber_shimplegako();
+			}
+		},
+
+		mouseinput_auto: function() {
+			if (this.puzzle.playmode) {
+				if (this.mousestart || this.mousemove) {
+					this.dragnumber_shimplegako();
+				}
+				if (this.mouseend && this.notInputted()) {
+					this.mouseCell = this.board.emptycell;
+					this.inputqnum();
+				}
+			}
+			if (this.puzzle.editmode && this.mousestart) {
+				this.inputqnum();
+			}
+		},
+
+		dragnumber_shimplegako: function() {
+			var cell = this.getcell();
+			if (cell.isnull || cell === this.mouseCell) {
+				return;
+			}
+
+			if (this.inputData === null) {
+				this.inputData = cell.getNum();
+				this.mouseCell = cell;
+				return;
+			}
+
+			if (this.inputData === -2) {
+				this.inputData = -1;
+			}
+
+			if (cell.qnum === -1) {
+				cell.clrSnum();
+				cell.setAnum(this.inputData);
+				cell.draw();
+			}
+			this.mouseCell = cell;
+		}
+	},
+
+	//---------------------------------------------------------
+	// キーボード入力系
+	KeyEvent: {
+		enablemake: true,
+		enableplay: true
+	},
+
+	//---------------------------------------------------------
+	// 盤面管理系
+	Cell: {
+		enableSubNumberArray: true,
+		maxnum: function() {
+			return this.board.cols + this.board.rows - 1;
+		}
+	},
+
+	Board: {
+		cols: 6,
+		rows: 6
+	},
+
+	//---------------------------------------------------------
+	// 画像表示系
+	Graphic: {
+		paint: function() {
+			this.drawBGCells();
+			this.drawTargetSubNumber();
+			this.drawGrid();
+
+			this.drawSubNumbers();
+			this.drawAnsNumbers();
+			this.drawQuesNumbers();
+
+			this.drawChassis();
+			this.drawCursor();
+		}
+	},
+
+	//---------------------------------------------------------
+	// URLエンコード/デコード処理
+	Encode: {
+		decodePzpr: function(type) {
+			this.decodeNumber16();
+		},
+		encodePzpr: function(type) {
+			this.encodeNumber16();
+		}
+	},
+
+	//---------------------------------------------------------
+	FileIO: {
+		decodeData: function() {
+			this.decodeCellQnum();
+			this.decodeCellAnumsub();
+		},
+		encodeData: function() {
+			this.encodeCellQnum();
+			this.encodeCellAnumsub();
+		}
+	},
+
+	//---------------------------------------------------------
+	// 正解判定処理実行部
+	AnsCheck: {
+		checklist: ["checkNoNumCell", "checkRowsColsSameNumber"],
+
+		checkRowsColsSameNumber: function() {
```

Comment:
Currently, this error doesn't show up until every cell is filled, which is too late in my opinion.

Can you split this check up in "too many numbers" and "not enough numbers", so we can put the "too many numbers" check in front of the empty cell check?